test/src-test/lib/fs-utils.spec.ts

```diff
@@ -0,0 +1,220 @@
+			await createTestFiles();
+		});
+
+		afterEach(async () => {
+			await removeTestFiles();
+		});
+
+		it('should write and sync string data', async () => {
+			await fsUtils.writeAndSyncFile(testFile1, 'foo bar');
+			expect(await fs.readFile(testFile1, 'utf-8')).to.equal('foo bar');
+		});
+
+		it('should write and sync buffers', async () => {
+			await fsUtils.writeAndSyncFile(testFile1, Buffer.from('bar foo'));
+			expect(await fs.readFile(testFile1, 'utf-8')).to.equal('bar foo');
+		});
+	});
+
+	describe('writeFileAtomic', () => {
+		before(async () => {
+			spy(fs, 'rename');
+			await createTestFiles();
+		});
+
+		after(async () => {
+			(fs.rename as SinonSpy).restore();
+			await removeTestFiles();
+		});
+
+		it('should write string data atomically', async () => {
+			await fsUtils.writeFileAtomic(testFile1, 'foo baz');
+			expect(await fs.readFile(testFile1, 'utf-8')).to.equal('foo baz');
+			expect(fs.rename).to.have.been.calledWith(`${testFile1}.new`, testFile1);
+		});
+
+		it('should write buffer data atomically', async () => {
+			await fsUtils.writeFileAtomic(testFile1, 'baz foo');
+			expect(await fs.readFile(testFile1, 'utf-8')).to.equal('baz foo');
+			expect(fs.rename).to.have.been.calledWith(`${testFile1}.new`, testFile1);
+		});
+	});
+
+	describe('safeRename', () => {
+		beforeEach(async () => {
+			await createTestFiles();
+		});
+
+		afterEach(async () => {
+			await removeTestFiles();
+		});
+
+		it('should rename a file', async () => {
+			await fsUtils.safeRename(testFile1, testFile1 + 'rename');
+			const dirContents = await fs
+				.readdir(fsUtilsRoot)
+				.then((dir) =>
+					dir.filter((files) => files.match(/temp-fs-utils-file\.[1|2]/)),
+				);
+			expect(dirContents).to.have.length(2);
+			expect(dirContents).to.include('temp-fs-utils-file.1rename');
+			expect(dirContents).to.not.include('temp-fs-utils-file.1');
+			await fs.unlink(testFile1 + 'rename');
+		});
+
+		it('should replace an existing file', async () => {
+			await fsUtils.safeRename(testFile1, testFile2);
+			const dirContents = await fs
+				.readdir(fsUtilsRoot)
+				.then((dir) =>
+					dir.filter((files) => files.match(/temp-fs-utils-file\.[1|2]/)),
+				);
+			expect(dirContents).to.have.length(1);
+			expect(dirContents).to.include('temp-fs-utils-file.2');
+			expect(dirContents).to.not.include('temp-fs-utils-file.1');
+		});
+	});
+
+	// This test suite can be reimplemented after we fix the testing situation. It's currently
+	// the absolute last test suite to be called and thus inherits the baggage
+	describe.skip('existsAsync', () => {
+		before(async () => {
+			await createTestFiles();
+		});
+
+		after(async () => {
+			await removeTestFiles();
+		});
+
+		it('should return whether a file exists', async () => {
+			expect(await fsUtils.existsAsync(testFile1)).to.be.true;
+			await fs.unlink(testFile1).catch(() => {
+				/* noop */
+			});
+			expect(await fsUtils.existsAsync(testFile1)).to.be.false;
+		});
+	});
```

Comment:
Probable reason for this failing is that fs.access might be flipping out because of invalid absolute directory

Reply:
It's not. We just don't remove the build folder properly it between test runs, so it's likely that an old file/script was interfering somehow. After doing `rm -rf build`, the tests passed.